cookbooks/bcpc/libraries/utils.rb

```diff
@@ -22,6 +22,17 @@
 require 'thread'
 require 'ipaddr'
 
+# create databag if missing
+def create_databag(name)
```

Comment:
I don't thin we need `create_databag` method. Also `init` method is not required anymore. The signature for `make_config` should change to `(key, item, bag)`  and `item` and `bag` should default to `node.chef_environment` and `configs` respectively. The `make_config` method should handle the logic for creation of `data bags` if it doesn't exist.